src/main/resources/db/migration/general/V14__hard_delete.sql

```diff
@@ -0,0 +1,97 @@
+alter table offender_case_note
+    rename column offender_case_note_id to id;
+
+alter table offender_case_note
+    rename column offender_identifier to person_identifier;
+
+alter table offender_case_note
+    rename column case_note_type_id to type_id;
```

Comment:
Realise we're a little in the middle of renaming. This is the case note's sub type in the code base and at some point it would make sense to rename the two reference data columns in line with that. Worth using `sub_type_id` in this PR?